docs/src/user_docs/assembly/io_operations.md

```diff
@@ -53,7 +53,7 @@ Memory is guaranteed to be initialized to zeros. Thus, when reading from memory
 | mem_loadw <br> - *(1 cycle)*  <br> mem_loadw.*a* <br> - *(2 cycles)*  | [a, 0, 0, 0, 0, ... ] | [A, ... ] | $A \leftarrow mem[a]$ <br> Reads a word from memory at address $a$ and overwrites top four stack elements with it. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
 | mem_store <br> - *(2 cycles)*  <br> mem_store.*a*  <br> - *(3-4 cycles)*   | [a, v, ... ] | [ ... ] | $v \rightarrow mem[a][0]$ <br> Pops the top element off the stack and stores it as the first element of the word in memory at address $a$. All other elements of the word are not affected. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
 | mem_storew <br> - *(1 cycle)*  <br> mem_storew.*a* <br> - *(2-3 cycles)*  | [a, A, ... ] | [A, ... ] | $A \rightarrow mem[a]$ <br> Stores the top four elements of the stack in memory at address $a$. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
-| mem_stream <br> - *(2 cycles)* | [S2, S1, S0, a, ...] | [T2, T1, T0, b, ...] | $[T_0, T_1, T_2] \leftarrow permute(S_0, mem[a], mem[a+1])$ <br> $b \leftarrow a + 2$ <br> Loads two words from memory starting at the address $a$, overwrites the top 8 elements of the stack with them, and applies Rescue Prime Optimized permutation to the top 12 elements of the stack. At the end of the operation the address is incremented by $2$. |
+| mem_stream <br> - *(2 cycles)* | [A, B, C, a, ...] | [A', B', C', a', ...] | $[A', B', C'] \leftarrow permute(C, mem[a], mem[a+1])$ <br> $a' \leftarrow a + 2$ <br> Loads two words (8 elements) from memory starting at the address $a$, overwrites these top 8 elements onto the operand stack, and performs a Rescue Prime Optimized permutation to the top 12 elements of the operand stack, where the first 8 elements are the rate and the last 4 the capacity. At the end of the operation the address is incremented by $2$. |
```

Comment:
nits: 
- the new usage of "overwrites" isn't quite correct here, because we're not overwriting these top 8 elements, we're overwriting the elements of the operand stack. We could say "overwrites the top 8 elements of the operand stack with the 2 words loaded from memory" or we could say something like "writes these 8 elements onto the top 8 positions of the operand stack, ..." or we could go for a different rephrase
- it should be "performs a Rescue Prime Optimized permutation on the top 12..."

Reply:
- replaced mentions of the data being copied to the operand stack, since that is an implementation detail from the MASM code (it is overwritten right after by the HPerm, and the intermediary state is never visible by the user)
- `to the top` -> `on the top`